In [1]:
import pandas
import scanpy
from pathlib import Path
import sys
import itertools
from matplotlib import pyplot
import numpy
from urllib import parse

In [2]:
HTSW = str(Path('~/proj/htsworkflow').expanduser())
if HTSW not in sys.path:
    sys.path.append(HTSW)
from htsworkflow.submission.encoded import ENCODED


In [3]:
server = ENCODED('www.encodeproject.org')
server.load_netrc()

In [4]:
libraries_used = set()
prefix_len = len('barbara-wold:')
with open('fastqs-used-in-c1-e10.5.txt', 'rt') as instream:
    for i, line in enumerate(instream):
        url = parse.urlparse(line.strip())
        parts = url.path.split('/')
        fastq_accession = parts[2]
        fastq = server.get_json(fastq_accession)
        library = fastq['library']
        library_id = library['@id']
        aliases = library['aliases']
        jumpgate = aliases[0][prefix_len:]
        #print(fastq_accession, library_id, jumpgate)
        libraries_used.add(jumpgate.split('_')[0])


In [5]:
libraries_used

{'17327',
 '17328',
 '17329',
 '17330',
 '17331',
 '17332',
 '17333',
 '17334',
 '18042',
 '18043',
 '18044',
 '18045',
 '18046',
 '18047',
 '18048',
 '18049',
 '18087',
 '18088',
 '18089',
 '18255',
 '18267',
 '18274',
 '18315',
 '19910',
 '19912',
 '19913',
 '20048'}

In [6]:
len(libraries_used)

27

Ran this through postgres

select  insert_size from samples_library  where id in ('17327', '17328', '17329', '17330', '17331', '17332', '17333', '17334', '18042', '18043', '18044', '18045', '18046', '18047', '18048', '18049', '18087', '18088', '18089', '18255', '18267', '18274', '18315', '19910', '19912', '19913', '20048');

which gave me the following insert sizes

In [7]:
insert_sizes = [ 662, 655, 569, 563, 603, 557, 493, 558, 362, 339, 327, 365, 338, 332, 309, 310, 472, 482, 554, 509, 266, 519, 387, 652, 555, 617, 527,]

In [8]:
numpy.std(insert_sizes)

120.92707730153474

In [9]:
numpy.mean(insert_sizes)

477.1111111111111